go/Makefile

```diff
@@ -171,10 +192,18 @@ long-timeout-test: __it
 opentelemetry-test: export TEST_FILTER = -skip "TestGlideTestSuite/TestModule" -otel-test $(if $(test-filter), -testify.m $(test-filter), -testify.m TestOpenTelemetry)
 opentelemetry-test: __it
 
+# Test commands for different platforms
+# musl gcc does not support -fsanitize=address, so we use clang instead
+# MUSL_TEST_CMD := CC=clang go test -asan -v ./integTest/...
+# TODO - Investigate musl + address sanitizer issue - https://github.com/valkey-io/valkey-glide/issues/4671
+# Disabled for now
+MUSL_TEST_CMD := CC="gcc" go test -v ./integTest/...
+REGULAR_TEST_CMD := CC="gcc -fsanitize=address" go test -v ./integTest/...
+
 __it:
 	mkdir -p reports
 	set -o pipefail; \
-	CC="gcc -fsanitize=address" go test -v ./integTest/... \
+	$(if $(filter true,$(IS_MUSL)),$(MUSL_TEST_CMD),$(REGULAR_TEST_CMD)) \
```

Comment:
Can we refactor this so that we test IS_MUSL first then set a single TEST_CMD variable rather than conditionally using MUSL_TEST_CMD vs. REGULAR_TEST_CMD? It's hard to read and keep track of this many variables in a single line.